### ISSUE TITLE
After client update becomes available and downloads, restart button doesn't work

### ISSUE THREAD

**Issue by tec27** [ShieldBattery/ShieldBattery#474]
The log shows some errors:

```
[2020-05-27T11:10:25.051Z]	<system>	Logging started
[2020-05-27T11:10:25.051Z]	<system>	Version: 6.1.3
[2020-05-27T11:10:25.072Z]	<info>	Checking for update
[2020-05-27T11:10:25.983Z]	<verbose>	Game server listening on port 65122
[2020-05-27T11:10:26.520Z]	<info>	Found version 6.1.4 (url: ShieldBattery-Dev-setup-6.1.4.exe)
[2020-05-27T11:10:26.520Z]	<info>	Downloading update from ShieldBattery-Dev-setup-6.1.4.exe
[2020-05-27T11:10:26.521Z]	<debug>	updater cache dir: C:\Users\Travis\AppData\Local\shieldbattery-app-updater
[2020-05-27T11:10:26.523Z]	<info>	No cached update info available
[2020-05-27T11:10:26.893Z]	<info>	Checking for update
[2020-05-27T11:10:27.077Z]	<info>	Found version 6.1.4 (url: ShieldBattery-Dev-setup-6.1.4.exe)
[2020-05-27T11:10:27.077Z]	<info>	Downloading update from ShieldBattery-Dev-setup-6.1.4.exe
[2020-05-27T11:10:27.078Z]	<info>	No cached update info available
[2020-05-27T11:10:28.461Z]	<error>	Cannot download differentially, fallback to full download: Error: Server doesn't support Accept-Ranges (response code 200)
    at o (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:509214)
    at ClientRequest.<anonymous> (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:147141)
    at ClientRequest.emit (events.js:210:5)
    at SimpleURLLoaderWrapper.<anonymous> (electron/js2c/browser_init.js:2510:12)
    at SimpleURLLoaderWrapper.emit (events.js:210:5)
[2020-05-27T11:10:28.544Z]	<error>	Cannot download differentially, fallback to full download: Error: Server doesn't support Accept-Ranges (response code 200)
    at o (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:509214)
    at ClientRequest.<anonymous> (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:147141)
    at ClientRequest.emit (events.js:210:5)
    at SimpleURLLoaderWrapper.<anonymous> (electron/js2c/browser_init.js:2510:12)
    at SimpleURLLoaderWrapper.emit (events.js:210:5)
[2020-05-27T11:10:33.841Z]	<info>	New version 6.1.4 has been downloaded to C:\Users\Travis\AppData\Local\shieldbattery-app-updater\pending\ShieldBattery-Dev-setup-6.1.4.exe
[2020-05-27T11:10:35.583Z]	<error>	Error: Error: ENOENT: no such file or directory, rename 'C:\Users\Travis\AppData\Local\shieldbattery-app-updater\pending\temp-ShieldBattery-Dev-setup-6.1.4.exe' -> 'C:\Users\Travis\AppData\Local\shieldbattery-app-updater\pending\ShieldBattery-Dev-setup-6.1.4.exe'
[2020-05-27T11:11:26.814Z]	<info>	Install on explicit quitAndInstall
[2020-05-27T11:11:26.814Z]	<error>	Error: Error: No valid update available, can't quit and install
    at y.install (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:141366)
    at y.quitAndInstall (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:140886)
    at IpcMainImpl.<anonymous> (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:218290)
    at IpcMainImpl.emit (events.js:210:5)
    at WebContents.<anonymous> (electron/js2c/browser_init.js:3851:15)
    at WebContents.emit (events.js:210:5)
[2020-05-27T11:11:47.980Z]	<info>	Install on explicit quitAndInstall
[2020-05-27T11:11:47.980Z]	<error>	Error: Error: No valid update available, can't quit and install
    at y.install (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:141366)
    at y.quitAndInstall (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:140886)
    at IpcMainImpl.<anonymous> (C:\Users\Travis\AppData\Local\Programs\ShieldBattery-Dev\resources\app.asar\index.js:2:218290)
    at IpcMainImpl.emit (events.js:210:5)
    at WebContents.<anonymous> (electron/js2c/browser_init.js:3851:15)
    at WebContents.emit (events.js:210:5)
```

**Comment by tec27** [comment]
After launching it again, and watching the mentioned folder as it downloaded, I saw it save the relevant files to that folder, and then within a few seconds they were all deleted. Unsure why as of yet.

**Comment by tec27** [comment]
Relevant code for the logs is here:
https://github.com/electron-userland/electron-builder/blob/3ffea83eaf399a6cab0db460f9777fea4e1631bd/packages/electron-updater/src/AppUpdater.ts#L617

Seems like we might be double-triggering the update somehow, and when the second one fails, it deletes all the files from the successful first one.

**Comment by tec27** [comment]
Should be fixed with abcf8a3bca2a3399445ea9d0ad94a0b719c5c55c (but we won't actually know until the version after this one)
